src/blog/2024/01/send-a-file.md

```diff
@@ -0,0 +1,166 @@
+import requests
+
+def send_file(nodered_url, file_path):
+    # Open the file in binary mode
+    with open(file_path, 'rb') as file:
+        files = {'file': (file.name, file, 'multipart/form-data')}
+        response = requests.post(nodered_url, files=files)
+
+    return response
+
+# Update the ip address and port of your Node-RED instance
+nodered_url = 'http://localhost:1880/fileupload'
+
+# Update the location of your file
+file_path = 'C:/Users/myUser/Downloads/shiftSchedule.csv'
+
+response = send_file(nodered_url, file_path)
+print(f"Response Status Code: {response.status_code}")
+print(f"Response Body: {response.text}")
+```
+
+Update the **nodered_url** to the location of the NodeRED instance.  Be sure to adjust the port if the default port of 1880 isn't being used.
+
+Update the **file_path** with the path to where the file to be uploaded will be located.
+
+**Save**
+
+To run:
+
+```python
+python run.py
+```
+
+
+## Stand Alone Web Application
+
+![csv upload application](./images/csv_upload_app.png)
+
+This stand-alone web application can be run on either Windows or Linux, .bat for Windows, and .sh for Linux.
+
+### Installation
+
+Clone the repository:
+```bash
+git clone https://github.com/gdziuba/FF_Send-File-to-NR.git
+```
+Navigate to the directory:
+```bash
+cd FF_Send-File-to-NR
+```
+
+### Configuration ###
+
+Edit the lines in the body of [index.html](https://github.com/gdziuba/FF_Send-File-to-NR/blob/21214f88c6c4536f49efb88cf5f84bf52071a88b/templates/index.html#L69) to include the endpoints to which you would like to send the files.
+
+```
+<option value="http://localhost:1880/fileupload">CSV File Upload</option>
+```
+
+### Operating Systems
+
+### Windows ###
+
+Run the script:
+```bash
+.\start_app.bat
+```
+
+This will install if necessary, start the Flask Application, and take you to localhost:5000 on the browser.
+
+### Linux ###
+
+Make the script executable: Run:
+
+```bash
+chmod +x setup_and_run.sh
+```
+
+To Run:
+
+```bash
+./setup_and_run.sh
+```
+
+To access the application, open a browser to the ip:5000 of the running application.
+
+
+
+### Node-RED Ingress
+
+<!-- ![csv upload application](./images/nr_flow_csv_ingress.png) -->
+
+<iframe width="100%" height="225px" src="https://flows.nodered.org/flow/effb53752e5d6f767b3c7e5d41a4a6e8/share?height=100" allow="clipboard-read; clipboard-write" style="border: none;"></iframe>
+
+Once we have a file ready to be sent, we now need to configure the receiving side in Node-RED. In this example, we are leveraging a CSV formatted file and then converting it to be used at a later time.
+
+A link to the flow can be found [here](https://flows.nodered.org/flow/effb53752e5d6f767b3c7e5d41a4a6e8).
+
+To import the flow, follow these [instructions](https://flowfuse.com/blog/2023/03/3-quick-node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import).
+
+A Simple HTTP In node can be used in the form of a Post, ensuring the configuration allows for a file.
```

Comment:
```suggestion
Once we have a file ready to be sent, we now need to configure the receiving side in Node-RED. In this example, we are leveraging a CSV formatted file and then converting it to be used at a later time. A Simple HTTP In node can be used in the form of a Post, ensuring the configuration allows for a file.

To import the flow, follow these [instructions](https://flowfuse.com/blog/2023/03/3-quick-node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import).
```

Reply:
the iframe already has a link to copy. :)

Reply:
It does, but I am unsure if everyone knows how to use that.  That is why I have both.